account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,821 @@
+            # TODO : on invoice creation, the lines are in the same
+            # order, but not on invoice update...
+            for i in range(len(parsed_inv['lines'])):
+                if 'price_subtotal' not in parsed_inv['lines'][i]:
+                    continue
+                iline = invoice.invoice_line_ids[i]
+                odoo_subtotal = iline.price_subtotal
+                parsed_subtotal = parsed_inv['lines'][i]['price_subtotal']
+                if float_compare(
+                        odoo_subtotal, parsed_subtotal,
+                        precision_rounding=prec):
+                    diff_amount = float_round(
+                        parsed_subtotal - odoo_subtotal,
+                        precision_rounding=prec)
+                    logger.info(
+                        'Price subtotal difference found on invoice line %d '
+                        '(source:%s, odoo:%s, diff:%s).',
+                        i + 1, parsed_subtotal, odoo_subtotal, diff_amount)
+                    copy_dict = {
+                        'name': _('Adjustment on %s') % iline.name,
+                        'quantity': 1,
+                        'price_unit': diff_amount,
+                    }
+                    if import_config['invoice_line_method'] ==\
+                            'nline_auto_product':
+                        copy_dict['product_id'] = False
+                    # Add the adjustment line
+                    iline.copy(copy_dict)
+                    logger.info('Adjustment invoice line created')
+        if float_compare(
+                parsed_inv['amount_untaxed'], invoice.amount_untaxed,
+                precision_rounding=prec):
+            # create global ajustment line
+            diff_amount = float_round(
+                parsed_inv['amount_untaxed'] - invoice.amount_untaxed,
+                precision_rounding=prec)
+            logger.info(
+                'Amount untaxed difference found '
+                '(source: %s, odoo:%s, diff:%s)',
+                parsed_inv['amount_untaxed'], invoice.amount_untaxed,
+                diff_amount)
+            il_vals = self._prepare_global_adjustment_line(
+                diff_amount, invoice, import_config)
+            il_vals['invoice_id'] = invoice.id
+            self.env['account.invoice.line'].create(il_vals)
+            logger.info('Global adjustment invoice line created')
+        # Invalidate cache
+        invoice = self.env['account.invoice'].browse(invoice.id)
+        assert not float_compare(
+            parsed_inv['amount_untaxed'], invoice.amount_untaxed,
+            precision_rounding=prec)
+        # Force tax amount if necessary
+        if float_compare(
+                invoice.amount_total, parsed_inv['amount_total'],
+                precision_rounding=prec):
+            if not invoice.tax_line_ids:
+                raise UserError(_(
+                    "The total amount is different from the untaxed amount, "
```

Comment:
I'm getting errors in this line because no tax lines are created before, could not find any error in the invoice file, in the part that refers to taxes. Am i missing something?

`<TaxesOutputs><Tax><TaxTypeCode>01</TaxTypeCode><TaxRate>21.00</TaxRate><TaxableBase><TotalAmount>885.00</TotalAmount><EquivalentInEuros>885.00</EquivalentInEuros></TaxableBase><TaxAmount><TotalAmount>185.85</TotalAmount><EquivalentInEuros>185.85</EquivalentInEuros></TaxAmount></Tax></TaxesOutputs><InvoiceTotals><TotalGrossAmount>885.00</TotalGrossAmount><TotalGrossAmountBeforeTaxes>885.00</TotalGrossAmountBeforeTaxes><TotalTaxOutputs>185.85</TotalTaxOutputs><TotalTaxesWithheld>0.00</TotalTaxesWithheld><InvoiceTotal>1070.85</InvoiceTotal><TotalOutstandingAmount>1070.85</TotalOutstandingAmount><TotalExecutableAmount>1070.85</TotalExecutableAmount></InvoiceTotals><Items><InvoiceLine><ItemDescription>[CARD] Graphics Card</ItemDescription><Quantity>1.0</Quantity><UnitPriceWithoutTax>885.000000</UnitPriceWithoutTax><TotalCost>885.000000</TotalCost><GrossAmount>885.000000</GrossAmount><TaxesOutputs><Tax><TaxTypeCode>01</TaxTypeCode><TaxRate>21.00</TaxRate><TaxableBase><TotalAmount>885.00</TotalAmount><EquivalentInEuros>885.00</EquivalentInEuros></TaxableBase><TaxAmount><TotalAmount>185.85</TotalAmount><EquivalentInEuros>185.85</EquivalentInEuros></TaxAmount></Tax></TaxesOutputs><ArticleCode>CARD</ArticleCode>`

Reply:
@aheficent I think you are facing at this issue: https://github.com/OCA/edi/pull/41#issuecomment-380708409